podaac/podaac.py

```diff
@@ -501,7 +507,28 @@ def granule_subset(self, input_file_path):
         token = result['token']
         conn.close()
 
-        return token
+        flag = 0
+        while(flag == 0):
+            url = url = self.URL + "subset/status?token=" + token
+            subset_response = requests.get(url).text
+            subset_response_json = json.loads(subset_response)
+            status = subset_response_json['status']
+            if (status == "done"):
+                flag = 1
+            if (status == "error"):
+                raise Exception(
+                    "Unexpected error occured for the subset job you have requested")
+            time.sleep(1)
+
+        print("Done! downloading the dataset zip .....")
+        download_url = subset_response_json['resultURLs'][0]
+        if path == '':
+            path = os.path.join(os.path.dirname(__file__), 'dataset.zip')
```

Comment:
Please grab the dataset name correctly from the ```download_url``` and append the .zip to this name.